packages/grafana-data/src/types/datasource.ts

```diff
@@ -373,6 +373,13 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /**
+   * Optionally, you can implement migrateQuery to handle query migrations when the query structure changes.
+   * Currently behind the feature flag: queryMigrations
+   * @alpha -- experimental
```

Comment:
can we specify when it runs and who calls this?  The two reasonable options are either dashboard load or editor UI load.  We should not call it for every query